commands/azureCommands/pull-from-azure.ts

```diff
@@ -38,19 +43,46 @@ export async function pullFromAzure(context?: AzureImageTagNode | AzureRepositor
     }
 
     // Using loginCredentials function to get the username and password. This takes care of all users, even if they don't have the Azure CLI
-    const credentials = await acrTools.loginCredentials(registry);
+    const credentials = await acrTools.getLoginCredentials(registry);
     const username = credentials.username;
     const password = credentials.password;
 
-    // Send commands to terminal
+    // Check if user is logged into Docker and send appropriate commands to terminal
     const terminal = ext.terminalProvider.createTerminal("Docker");
     terminal.show();
+
     let cont = (err, stdout, stderr) => {
         ext.outputChannel.append(stdout);
         ext.outputChannel.append(stderr);
-        terminal.sendText(`docker pull ${registryName}/${imageName}`);
+    };
+
+    if (!isLoggedIntoDocker(registry)) {
+        try {
+            exec(`docker login ${registryName} -u ${username} -p ${password}`, cont);
+        } catch (error) {
+            console.log(error);
+        }
+        // exec(`docker login ${registryName} -u ${username} -p ${password}`, (err, stdout, stderr) => {
+        //     ext.outputChannel.append(stdout);
+        //     ext.outputChannel.append(stderr);
+        // });
+        console.log("*was not logged into docker, just completed dockerlogin")
     }
+    terminal.sendText(`docker pull ${registryName}/${imageName}`);
+}
 
-    exec(`docker login ${registryName} -u ${username} -p ${password}`, cont);
+function isLoggedIntoDocker(registry: Registry): boolean {
+    let home = process.env.HOMEPATH;
+    let configPath: string = path.join(home, '.docker', 'config.json');
+    let buffer: Buffer;
+    try {
```

Comment:
Get rid of the try/catch and do it this way, which will give us telemetry if (and only if) it fails (otherwise we won't know if anything wrong is happening on users' machines):
```
        await callWithTelemetryAndErrorHandling('findDockerConfig', async function (this: IActionContext): Promise<void> {
            this.suppressTelemetry = true;

            buffer = fs.readFileSync(configPath); 
});
```